packages/jsdoc-plugins/lib/validator/doclet-validator.js

```diff
@@ -316,6 +312,9 @@ class DocletValidator {
 	 * @returns {Boolean}
 	 */
 	_isCorrectType( type ) {
+		// JSDoc converts `Type.<Function>` to `Type.<function()>` for some reason...
+		type = type.replace( 'function()', 'function' );
```

Comment:
JSDoc converts `Function` param to `function()` on its own. I don't know why at the moment.
`type = type.replace( 'function()', 'function' );` - this is to avoid errors thrown by `doclet-validator`